### DISCUSSION TITLE
How do I create a pdf figure generated by R

### DISCUSSION THREAD

**Discussion by olechnwin** [quarto-dev/quarto-cli#7052] [Q&A] [answered]
### Description

I tried to create a pdf figure using fig-format.

```
---
title: "test"
format: html
editor: visual
---

```{r}
#| fig-format: pdf
plot(1 + 1)
```

The html has the figure. I was expecting to see a pdf figure in test_files/ but there is only png figure.

```
ls test_files/figure-html/
unnamed-chunk-1-1.png
```
How do I create a pdf figure of plot(1+1) without using `pdf` function?

**Comment by mcanouil** [comment]
In R, figures are handled by "graphic devices", you have to use those, see <https://bookdown.org/yihui/rmarkdown-cookbook/graphical-device.html>

````qmd
---
title: "test"
format: html
---

```{r}
#| dev: pdf
plot(1 + 1)
```
````

---

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````

**Reply by olechnwin** [reply]
@mcanouil,

Thank you for your reply and for the tips on posting a Quarto document.

I tried using `dev:pdf` but the embedded figure becomes like this:

![image](https://github.com/quarto-dev/quarto-cli/assets/20069067/17ca635c-8125-4e85-881d-68679662a25c)
Is there a way to make it look like what it was without the box around the figure?

**Reply by mcanouil** [reply]
You can't have a pdf appears as a png in HTML.

`dev` accepts multiple devices, so you can make the code cell generate several image files, see <https://yihui.org/knitr/options/>.

**Reply by olechnwin** [reply]
Thank you! This gets me what I wanted.
````
```{r, dev=c("png","pdf")}
plot(1 + 1)
```
````
But, how do I do this with the `#| ` ? I tried the following but it gave me an error:
````
```{r}
#| dev: c("png","pdf")
plot(1 + 1)
```
````

**Reply by cderv** [reply, marked as answer]
Two things: 

* `fig-format` should work as this is an alias for `dev`. Though this requires latest **knitr** and possibly Quarto 1.4. So in the meantime using `dev` is fine (but this options is only working for **knitr** chunks in .qmd)
* You need to use YAML syntax so use List syntax when multiple item. 

````markdown
---
title: Hello World
---

```{r}
#| fig-format:
#|   - png
#|   - pdf
plot(1 + 1)
```
````

This will indeed produce a png AND a pdf file, and include only the first I believe.

**Reply by olechnwin** [reply]
Thank you so much!
